search_service/api/table.py

```diff
@@ -115,3 +117,116 @@ def get(self, *, field_name: str, field_value: str) -> Iterable[Any]:
 
             err_msg = 'Exception encountered while processing search request'
             return {'message': err_msg}, HTTPStatus.INTERNAL_SERVER_ERROR
+
+
+class SearchTableFilterAPI(Resource):
+    """
+    Search Table API using search filtering
+
+    This API should be generic enough to support every search filter use case.
+    Todo: we should deprecate the SearchTableFieldAPI with this API as that one
+    only provides a subset of support.
+    """
+
+    def __init__(self) -> None:
+        self.proxy = get_proxy_client()
+
+        self.parser = reqparse.RequestParser(bundle_errors=True)
+
+        self.parser.add_argument('page_index', required=False, default=0, type=int)
+        self.parser.add_argument('index', required=False, default=TABLE_INDEX, type=str)
+
+        super(SearchTableFilterAPI, self).__init__()
+
+    @marshal_with(search_table_results)
+    @swag_from('swagger_doc/table/search_table_filter.yml')
+    def get(self) -> Iterable[Any]:
+        """
+        Fetch search results based on filter.
+
+        The frontend will send the generic json payload which will get converted
+        into query DSL depending on search backend engine(only support elasticsearch currently)
+
+        e.g
+        ```
+        {
+            'search_request': {
+                'type': 'AND'
+                filters: {
+                    'database': ['hive', 'bigquery'],
+                    'schema': ['test-schema1', 'test-schema2'],
+                    'table': ['*amundsen*'],
+                    'column': ['*ds*']
+                    'tag': ['test-tag']
+                }
+            }
+        }
+
+        This generic JSON will convert into DSL depending on the backend engines.
+
+        E.g in Elasticsearch, it will become
+        'database':('hive' OR 'bigquery') AND
+        'schema':('test-schema1' OR 'test-schema2') AND
+        'table':('*amundsen*') AND
+        'column':('*ds*') AND
+        'tag':('test-tag')
+        ```
+        :return: list of table results. List can be empty if query
+        doesn't match any tables
+        """
+        args = self.parser.parse_args(strict=True)
+
+        try:
+            search_request = json.loads(request.json).get('search_request')
+        except Exception:
+            msg = 'The search request payload is not available in the request'
```

Comment:
Should it be `400` bad request as it fails to parse?

Reply:
Updating this to return `HTTPStatus.BAD_REQUEST`